### PR TITLE
OS-8319 lx centos family resolver logic should match ubuntu

### PR DESCRIPTION
**Pull request by papertigers**

### CODE REVIEW

File: usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh [hunk 1]
```diff
@@ -59,44 +59,30 @@ case "\$1" in
...
 
-# Alter resolv.conf if we're pulling resolver information from zonecfg(1M).
-zonecfg -z $ZONENAME info attr name=resolvers | grep -q resolvers
-if [[ $? == 0 ]]; then
-    cat >> $tmpfile <<EOF
-    if [ ! -e /etc/resolv.conf ]; then
-        if [ -h /etc/resolv.conf ]; then
-            rm -f /etc/resolv.conf
+# Populate resolv.conf setup files IFF we have resolvers information.
+resolvers=`zone_attr resolvers`
```
Comment (bahamat) [inline usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh:63]: Prefer `$()`

File: usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh [hunk 2]
```diff
@@ -59,44 +59,30 @@ case "\$1" in
...
+if [ -n "$resolvers" ]; then
+
+    echo "# AUTOMATIC ZONE CONFIG" > $tmpfile
+    _IFS=$IFS; IFS=,; for r in $resolvers; do
+        echo "nameserver $r"
+    done >> $tmpfile
+    IFS=$_IFS
+    domain=`zone_attr dns-domain`
```
Comment (bahamat) [inline usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh:71]: Prefer `$()`

**Review by bahamat — CHANGES_REQUESTED** [review]
Will also need to update copyright. `make check` should complain about that.

**Comment by danmcd** [comment]
Needs testing notes on OS-8319 ticket.

**Review by danmcd — COMMENTED** [review]
One other question... will this break older Centos?  Do we care if it does?

**Review by danmcd — APPROVED** [review]
Comment about older Centos addressed by Brian on Slack.  Don't forget test notes!

**Review by bahamat — APPROVED** [review]

File: usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh [hunk 3]
```diff
@@ -59,44 +59,30 @@ case "\$1" in
...
 EOF
 
-# Alter resolv.conf if we're pulling resolver information from zonecfg(1M).
-zonecfg -z $ZONENAME info attr name=resolvers | grep -q resolvers
-if [[ $? == 0 ]]; then
-    cat >> $tmpfile <<EOF
-    if [ ! -e /etc/resolv.conf ]; then
-        if [ -h /etc/resolv.conf ]; then
-            rm -f /etc/resolv.conf
+# Populate resolv.conf setup files IFF we have resolvers information.
```
Comment (sjorge) [inline usr/src/lib/brand/lx/zone/lx_boot_zone_redhat.ksh:62]: IFF?

**Comment by danmcd** [comment]
"IFF" == shorthand for "if and only if".

**Comment by sjorge** [comment]
> "IFF" == shorthand for "if and only if".

I see, never seen that before! TIL

**Comment by danmcd** [comment]
Yeah, college/uni-level math classes and/or theoretical Computer Science drilled that into my head.  It also has formal logic implications:  a  -->  b   (If A then B), a <-- b, (A only if B), and a <--> b (A if and only if B, aka A IFF B).

**Pull request merged**